pkg/contextstore/metadata_test.go

```diff
@@ -0,0 +1,119 @@
+package contextstore
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+func TestMetadataMarshalling(t *testing.T) {
+	var ctxMeta ContextMetadata
+	expected := ContextMetadata{
+		Metadata: map[string]interface{}{
+			"baz": "foo",
+		},
+		Endpoints: map[string]EndpointMetadata{
+			"test-ep": {
+				"foo": "bar",
+			},
+		},
+	}
+	bytes, err := json.Marshal(&expected)
+	assert.NilError(t, err)
+	err = json.Unmarshal(bytes, &ctxMeta)
+	assert.NilError(t, err)
+	assert.DeepEqual(t, ctxMeta, expected)
+}
+
+func TestMetadataGetNotExisting(t *testing.T) {
+	testDir, err := ioutil.TempDir("", "TestMetadataGetNotExisting")
+	assert.NilError(t, err)
+	defer os.RemoveAll(testDir)
+	testee := metadataStore{root: testDir}
+	_, err = testee.get("noexist")
+	assert.Equal(t, true, os.IsNotExist(err))
+}
+
+func TestMetadataCreateGetRemove(t *testing.T) {
+	testDir, err := ioutil.TempDir("", "TestMetadataCreateGet")
+	assert.NilError(t, err)
+	defer os.RemoveAll(testDir)
+	testee := metadataStore{root: testDir}
+	expected := ContextMetadata{
```

Comment:
nit: You may factorize this expected ContextMetadata with TestMetadataMarshalling test